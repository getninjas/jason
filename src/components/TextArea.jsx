import React from 'react';
import PropTypes from 'prop-types';

export default class TextArea extends React.Component {
  static defaultProps = {
    id: '',
    placeholder: '',
    name: '',
    title: '',
    required: false,
    value: '',
  }

  render() {
    return (
      <textarea
        id={this.props.id}
        name={this.props.name}
        title={this.props.title}
        className="form__input"
        placeholder={this.props.placeholder}
        required={this.props.required}
        value={this.props.value} />
    );
  }
}

TextArea.propTypes = {
  id: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  placeholder: PropTypes.string,
  required: PropTypes.bool,
  value: PropTypes.any,
}
