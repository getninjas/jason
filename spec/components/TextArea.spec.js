import React from 'react';
import renderer from 'react-test-renderer';
import TextArea from '../../src/components/TextArea';

describe('TextArea', () => {
  it('renders custom props', () => {
    const component = renderer.create(
      <TextArea id={'idTest'}
        placeholder={'placeholderTest'}
        required={false}
        value={'ola test value'}
      />,
    );

    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders defaultProps', () => {
    const component = renderer.create(<TextArea />);

    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
