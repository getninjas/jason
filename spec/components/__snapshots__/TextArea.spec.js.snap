// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextArea renders defaultProps 1`] = `
<textarea
  className="form__input"
  id="idTest"
  maxLength={255}
  minLength={3}
  name="nameTest"
  onBlur={[Function]}
  onChange={[Function]}
  placeholder="placeholderTest"
  required={false}
  title=""
  value=""
/>
`;
